java/avro-converter/src/main/java/com/microsoft/AvroConverterConfig.java

```diff
@@ -0,0 +1,129 @@
+package com.microsoft;
+
+import java.util.HashMap;
+import java.util.Map;
+import com.azure.core.credential.TokenCredential;
+
+public class AvroConverterConfig extends KafkaAvroSerializerConfig {
+  public AvroConverterConfig(Map<String, ?> props) {
+    super((Map<String, Object>) props);
+  }
+
+  public static class Builder {
+    private Map<String, Object> props = new HashMap<String, Object>();
+
+    public Builder with(String key, Object value) {
+      props.put(key, value);
+      return this;
+    }
+
+    public AvroConverterConfig build() {
+      return new AvroConverterConfig(props);
+    }
+  }
+}
+
+class KafkaAvroSerializerConfig extends AbstractKafkaSerdeConfig {
+  /**
+   * If specified true, serializer will register schemas against Azure Schema Registry service under the specified
+   * group.  See Azure Schema Registry documentation for a description of schema registration behavior.
+   *
+   * If specified false, serializer will simply query the service for an existing ID given schema content.
+   * Serialization will fail if the schema has not been pre-created.
+   *
+   * Auto-registration is **NOT RECOMMENDED** for production scenarios.
+   *
+   * Requires type String.
+   */
+  public static final String AUTO_REGISTER_SCHEMAS_CONFIG = "auto.register.schemas";
+
+  public static final String AUTO_REGISTER_SCHEMAS_CONFIG_DEFAULT = "false";
+
+  public static final String AVRO_SPECIFIC_READER_CONFIG = "specific.avro.reader";
+
+  public static final Boolean AVRO_SPECIFIC_READER_CONFIG_DEFAULT = false;
+
+  public static final String AVRO_SPECIFIC_VALUE_TYPE_CONFIG = "specific.avro.value.type";
+
+  /**
+   * Specifies schema group for interacting with Azure Schema Registry service.
+   *
+   * If auto-registering schemas, schema will be stored under this group.
+   * If not auto-registering, serializer will request schema ID for matching data schema under specified group.
+   */
+  public static final String SCHEMA_GROUP_CONFIG = "schema.group";
+
+  KafkaAvroSerializerConfig(Map<String, Object> props) {
+      super(props);
+  }
+
+  /**
+   * @return auto-registration flag, with default set to false
+   */
+  public Boolean getAutoRegisterSchemas() {
```

Comment:
do we actually need the ability to auto-register schema in the Kafka Connect scenario? unless Kafka Connect at the most generic level require us to support it, I would suggest not providing this ability until someone ask for it. Auto-Registering is really not the best practice in production and customer can easily shoot themselves in the foot, so to speak.